tests/mvrf/test_mgmtvrf.py

```diff
@@ -172,20 +174,28 @@ def basic_check_after_reboot(self, duthost, localhost, testbed):
         inbound_test.test_ping(duthost, localhost)
         inbound_test.test_snmp_fact(localhost)
 
-    def test_warmboot(self, duthost, localhost, testbed):
-        duthost.command("sudo config save -y")
+    @pytest.fixture(scope="class")
+    def backup_restore_config(self, duthost):
+        # Backup the original config_db without mgmt vrf config
+        duthost.shell("cp /etc/sonic/config_db.json /etc/sonic/config_db.json.bak")
+        yield
+        # Restore the original config_db to override the config_db with mgmt vrf config
+        duthost.shell("mv /etc/sonic/config_db.json.bak /etc/sonic/config_db.json")
```

Comment:
Do we need to do a `config reload` to restore the original config_db once the test is over?

Reply:
Sure. Pushed a new commit:
* Added config reload in teardown
* Added back the code for getting free_port, which was accidentally removed while creating PR.

Reply:
Would there be race condition between config restore and config reload?

Even there is no race, I think putting them together could prevent future human error.

Reply:
Updated. I also changed `pytest_assert` to `raise Exception` in the `verify_show_command` function. Then we can have a exception to catch if something is wrong. Otherwise, pytest_assert may just fail the test and there is no exception to catch. This would make it difficult to insert the logic of restoring config_db.
Now, if configuring mgmt vrf failed somehow, the exception can always be caught. The code for restoring config_db won't b missed. The pytest.fail after that will just fail the whole test and stop executing rest of the test cases and teardown part.

Reply:
Yeah I think that looks good!